### PR TITLE
lloc: Count declaration statements for Cpp code

### LINKED ISSUE #614: Wrong lloc count
```
void XPCJSRuntime::CustomGCCallback(JSGCStatus status) {
  nsTArray<xpcGCCallback> callbacks(extraGCCallbacks.Clone());
  for (uint32_t i = 0; i < callbacks.Length(); ++i) {
    callbacks[i](status);
  }
}
```

lloc for this snippet is 1.0.

### PR DESCRIPTION
**Pull request by Luni-4**
This PR fixes #614 considering declaration statements for `Cpp` code

### CODE REVIEW

File: src/metrics/loc.rs [hunk 1]
```diff
@@ -880,6 +889,20 @@ mod tests {
...
+            "nsTArray<xpcGCCallback> callbacks(extraGCCallbacks.Clone());
+             for (uint32_t i = 0; i < callbacks.Length(); ++i) {
+                 callbacks[i](status);
+             }",
+            "foo.cpp",
+            CppParser,
+            loc,
+            [(lloc, 3, usize)] // nsTArray, for, callbacks
+        );
+    }
```
Comment (marco-c) [inline src/metrics/loc.rs:904]: Could you run the minimal tests comparison? Then after checking the changes are good, add a few more tests here.

**Comment by Luni-4** [comment]
Only improvements https://community-tc.services.mozilla.com/api/queue/v1/task/dLI5m6mhRUWM94kx5zUA6w/runs/0/artifacts/public%2Fjson-diffs-and-minimal-tests.tar.gz

File: src/metrics/loc.rs [hunk 2]
```diff
@@ -880,6 +889,60 @@ mod tests {
...
+    fn cpp_for_lloc() {
+        check_metrics!(
+            "for (; start != end; ++start) {
+                 const unsigned char idx = *start;
+                 if (idx > 127 || !kValidTokenMap[idx]) return false;
+             }",
+            "foo.cpp",
+            CppParser,
+            loc,
+            [(lloc, 4, usize)] // for, idx, if, return
```
Comment (marco-c) [inline src/metrics/loc.rs:928]: Shouldn't these be actually 3? The return is on the same line as the if

Comment (Luni-4) [reply src/metrics/loc.rs:928]: `lloc` counts by statements, so it's correct. An `if` is a statement and a `return` another one

**Review by marco-c — APPROVED** [review]

**Pull request merged**